server/src/main/java/org/elasticsearch/snapshots/SnapshotsService.java

```diff
@@ -1520,23 +1520,10 @@ public void clusterStateProcessed(String source, ClusterState oldState, ClusterS
 
         final Set<Index> indices = new HashSet<>();
         for (final SnapshotsInProgress.Entry entry : snapshots.entries()) {
-            if (entry.partial() == false) {
-                if (entry.state() == State.INIT) {
-                    for (IndexId index : entry.indices()) {
-                        IndexMetaData indexMetaData = currentState.metaData().index(index.getName());
-                        if (indexMetaData != null && indicesToCheck.contains(indexMetaData.getIndex())) {
-                            indices.add(indexMetaData.getIndex());
-                        }
-                    }
-                } else {
-                    for (ObjectObjectCursor<ShardId, SnapshotsInProgress.ShardSnapshotStatus> shard : entry.shards()) {
-                        Index index = shard.key.getIndex();
-                        if (indicesToCheck.contains(index)
-                            && shard.value.state().completed() == false
-                            && currentState.getMetaData().index(index) != null) {
-                            indices.add(index);
-                        }
-                    }
+            for (IndexId index : entry.indices()) {
```

Comment:
Admittedly this makes it a little "harder" to delete an index, but I don't see it as much of an issue relative to the complication it saves. If we don't do it this way, we'd have to add another step to write index metadata per index (once all the shards for that index have sucessfull been snapshotted) to the state machine which doesn't seem worth it?

Reply:
👍 I don't think this weakens anything meaningful, since the order in which indices are snapshotted isn't specified. The existing behaviour seems overly heroic.